great_expectations/experimental/datasources/pandas_datasource.py

```diff
@@ -62,14 +62,16 @@ def _fully_specified_batch_requests_with_path(
         option_to_group_id = self._option_name_to_regex_group_id()
         group_id_to_option = {v: k for k, v in option_to_group_id.items()}
         batch_requests_with_path: List[Tuple[BatchRequest, pathlib.Path]] = []
-        for filename in os.listdir(self.path):
-            match = self.regex.match(filename)
+
+        file_name: str
+        for file_name in os.listdir(self.path):
+            match = self.regex.match(file_name)
```

Comment:
```suggestion
        for filepath in self.path.iterdir():
            match = self.regex.match(filepath.name)
```

If you are modifying this code can we go ahead and get rid of the `os.path` usage.

Reply:
@Kilo59 Potentially -- I will look -- in case not too much depends on it.  Thanks.

Reply:
Thanks, and I should have said... this is a non-blocking request.
I'm okay to leave it as is.